src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -382,6 +396,20 @@ public override SyntaxNode AsPrivateInterfaceImplementation(SyntaxNode declarati
             });
         }
 
+        private SyntaxNode WithoutConstraints(SyntaxNode declaration)
+        {
+            if (declaration.IsKind(SyntaxKind.MethodDeclaration))
+            { 
+                var method = (MethodDeclarationSyntax)declaration;
+                if (method.ConstraintClauses.Count > 0)
+                {
+                    return this.RemoveNodes(method, method.ConstraintClauses);
```

Comment:
i.e. 'RemoveNodes' should check if the node list is empty, and just immediately bail if it is.